File: code/jvm/group7/src/main/kotlin/group7/daw/project/models/Project.kt
package group7.daw.project.models

import group7.daw.project.data.Transition

data class Project (
        val user_id: Int,
        val project_id: Int,
        val name: String,
        val description: String,
        var labels: MutableList<String> = mutableListOf(),
        var states: MutableList<String> = mutableListOf(),
        var transitions: MutableList<Transition> = mutableListOf()
        )

File: code/js/group7/src/common/types/project.ts
import { ActionsSiren } from "./actions";
import { LinksSiren } from "./links";
import { UserProperty } from "./user";

export type ProjectEntity = {
  class: Array<string>;
  rel: Array<string>;
  properties: UserProperty;
  links: LinksSiren;
};

export type ProjectProperty = {
  name: string;
  description: string;
  project_id: number;
  labels: Array<String>;
  states: Array<String>;
  transitions: Array<Transition>;
};

export type Transition = {
  project_id: number;
  state_name: string;
  transits_to: string;
};

export type ProjectType = {
  class: string;
  properties: ProjectProperty;
  entities: Array<ProjectEntity>;
  actions: Array<ActionsSiren>;
  links: Array<LinksSiren>;
};
